code/game/area/areas.dm

```diff
@@ -490,3 +497,130 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 /// Called when a living mob that spawned here, joining the round, receives the player client.
 /area/proc/on_joining_game(mob/living/boarder)
 	return
+
+// Day night stuff
+/**
+ * This will calculate all adjacent turfs in this area and add them to a cache for lighting effects.
+ *
+ * WARNING: This proc is VERY expensive and should be used sparingly.
+ */
+/area/proc/initialize_day_night_adjacent_turfs()
+	LAZYCLEARLIST(adjacent_day_night_turf_cache)
+	LAZYINITLIST(adjacent_day_night_turf_cache)
+
+	for(var/turf/iterating_turf in contents)
+		var/bitfield = NONE
+		for(var/bit_step in ALL_JUNCTION_DIRECTIONS)
+			var/turf/target_turf
+			switch(bit_step)
+				if(NORTH_JUNCTION)
+					target_turf = locate(iterating_turf.x, iterating_turf.y + 1, iterating_turf.z)
+				if(SOUTH_JUNCTION)
+					target_turf = locate(iterating_turf.x, iterating_turf.y - 1, iterating_turf.z)
+				if(EAST_JUNCTION)
+					target_turf = locate(iterating_turf.x + 1, iterating_turf.y, iterating_turf.z)
+				if(WEST_JUNCTION)
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y, iterating_turf.z)
+				if(NORTHEAST_JUNCTION)
+					if(bitfield & NORTH_JUNCTION || bitfield & EAST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x + 1, iterating_turf.y + 1, iterating_turf.z)
+				if(SOUTHEAST_JUNCTION)
+					if(bitfield & SOUTH_JUNCTION || bitfield & EAST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x + 1, iterating_turf.y - 1, iterating_turf.z)
+				if(SOUTHWEST_JUNCTION)
+					if(bitfield & SOUTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y - 1, iterating_turf.z)
+				if(NORTHWEST_JUNCTION)
+					if(bitfield & NORTH_JUNCTION || bitfield & WEST_JUNCTION)
+						continue
+					target_turf = locate(iterating_turf.x - 1, iterating_turf.y + 1, iterating_turf.z)
+			if(!target_turf)
+				continue
+			var/area/target_area = target_turf.loc
+			if(target_area == src)
+				continue
+			if(!target_area.outdoors || target_area.underground)
+				continue
+			bitfield ^= bit_step
+
+		if(!bitfield)
+			continue
+		adjacent_day_night_turf_cache[iterating_turf] = list(DAY_NIGHT_TURF_INDEX_BITFIELD, DAY_NIGHT_TURF_INDEX_APPEARANCE)
+		adjacent_day_night_turf_cache[iterating_turf][DAY_NIGHT_TURF_INDEX_BITFIELD] = bitfield
+		RegisterSignal(iterating_turf, COMSIG_PARENT_QDELETING, .proc/clear_adjacent_turf)
+		if(iterating_turf.lighting_object)
+			iterating_turf.lighting_object.day_night_area = src
```

Comment:
What exactly is the logic of this loop?

Reply:
Go through all turfs in the area contents, check N, NE, E, SE, S, SW, W, NW directions for adjacent turfs, check those turfs to see if they should be included in our adjacency overlays, and if so, add them to the list. In practice, it's not that expensive, but people shouldn't be calling this every second.

Reply:
why not use orange(1) for this?
Also, you know that iterating the contents of an area is equivalent in cost to iterating world, and filtering for "things" in that area, yes?